dev-java/batik/batik-1.8-r2.ebuild

```diff
@@ -0,0 +1,110 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Header: /var/cvsroot/gentoo-x86/dev-java/batik/batik-1.8-r1.ebuild,v 1.4 2015/06/28 14:58:49 monsieurp Exp $
+
+EAPI=5
+JAVA_PKG_IUSE="doc"
+
+inherit java-pkg-2 java-ant-2 eutils
+
+DESCRIPTION="Java based SVG toolkit"
+HOMEPAGE="https://xmlgraphics.apache.org/batik/"
+SRC_URI="http://apache.mirrors.ovh.net/ftp.apache.org/dist/xmlgraphics/${PN}/source/${PN}-src-${PV}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="1.8"
+KEYWORDS="~amd64 ~x86 ~ppc ~ppc64 ~amd64-linux ~x86-linux ~ppc-macos ~x86-macos ~sparc-solaris ~x86-solaris"
+IUSE="doc python tcl"
+
+CDEPEND="dev-java/xalan:0
+	dev-java/rhino:1.6
+	dev-java/xml-commons-external:1.3
+	dev-java/xmlgraphics-commons:2
+	python? ( dev-java/jython:0 )
+	tcl? ( dev-java/jacl:0 )
+	dev-java/ant-core:0"
+DEPEND=">=virtual/jdk-1.6
+	${CDEPEND}"
+RDEPEND=">=virtual/jre-1.6
+	${CDEPEND}"
+
+JAVA_ANT_REWRITE_CLASSPATH="true"
+EANT_GENTOO_CLASSPATH="
+	xml-commons-external-1.3
+	xmlgraphics-commons-2
+	xalan
+	rhino-1.6
+"
+
+java_prepare() {
+	# bug #318323
+	for file in build.xml contrib/rasterizertask/build.xml; do
+		java-ant_xml-rewrite -f ${file} -c -e javadoc -a failonerror -v no -a maxmemory -v 512m
+	done
+
+	# Add some missing imports to avoid a compiling issue.
+	# https://bugs.gentoo.org/show_bug.cgi?id=551952
+	# https://issues.apache.org/jira/browse/BATIK-1123
+	local imports=()
+	imports+=(sources/org/apache/batik/script/jpython/JPythonInterpreterFactory.java)
+	imports+=(sources/org/apache/batik/script/jacl/JaclInterpreterFactory.java)
+	for import in ${imports[@]}; do
+		einfo "Fixing missing import in ${import}"
+		sed -i '23i import org.apache.batik.script.ImportInfo;' ${import}
+		eend $?
+	done
+
+	cd lib
```

Comment:
Thanks for the careful review of the ebuild.  I left a note for -r1::gentoo here:
https://bugs.gentoo.org/show_bug.cgi?id=555870